### PR TITLE
Fix local-remote issue from release

### PR DESCRIPTION
**Pull request by vladaionescu**
Revert most of https://github.com/earthly/earthly/pull/810 as it was the issue here.

Add some better logic about what platform to use during export.

### CODE REVIEW

File: ast/parser/Earthfile [hunk 1]
```diff
@@ -8,16 +8,15 @@ RUN apk add --update --no-cache \
 
 # Install antlr.
 WORKDIR /usr/local/lib
-ARG ANTLR_VERSION=4.9.1
```
Comment (alexcb) [inline ast/parser/Earthfile:11]: Is this change intentional? now if we ever want to change the version we need to update it in two locations instead of just this one place. Alternatively maybe we can have curl output to just `antlr-complete.jar` so we only need to hardcode the version in one place.

File: llbutil/platform.go [hunk 2]
```diff
@@ -64,12 +65,33 @@ func PlatformToString(p *specs.Platform) string {
...
+	if p1 == nil {
+		return p2, nil
 	}
-	return override
+	if p2 == nil {
+		return p1, nil
+	}
+	plat1 := platforms.Normalize(*p1)
+	plat2 := platforms.Normalize(*p2)
+	if plat1.OS != plat2.OS {
```
Comment (alexcb) [inline llbutil/platform.go:79]: nice additional error checking here :)

**Review by alexcb — COMMENTED** [review]
Code looks good, just not sure about the `4.9.1` hardcoded value appearing twice.

File: ast/parser/Earthfile (hunk 1 above)
Comment (vladaionescu) [reply ast/parser/Earthfile]: This was part of the revert. Anyway, fixed now.

**Review by alexcb — APPROVED** [review]

File: llbutil/platform.go (hunk 2 above)
Comment (vladaionescu) [reply llbutil/platform.go:79]: Also part of the revert :-)

**Review by dchw — APPROVED** [review]

File: llbutil/platform.go (hunk 2 above)
Comment (alexcb) [reply llbutil/platform.go:79]: hah, nevermind me :)
